### ISSUE TITLE
Followers design/layout needs improvement

### ISSUE THREAD

**Issue by gabek** [owncast/owncast#3248]
### Share your bug report, feature request, or comment.

At its best it looks bad. At its worst it's really bad. Anybody have any ideas to improve it?

![image](https://github.com/owncast/owncast/assets/414923/de6f5510-1e68-4444-9d73-72cf88c69a47)

**Comment by thezeeshann** [comment]
@gabek I would like to work on this issue. assign me

**Comment by DineshDhanji** [comment]
Hi @gabek! I would like to work on this issue. I have also attached my idea. 
#### Suggestion
![Suggestion](https://github.com/owncast/owncast/assets/56204876/6b67b384-5a8d-4ecc-80b4-8f46ffcd7e13)

In case of longer name or text, we can truncate it (Like the second picture).

**Comment by gabek** [comment]
It does look better! But it would truncate for pretty much everyone, since accounts are almost always going to be longer than that. So if it's going to truncate for everybody, I'm not sure if that's the right design. What do you think?

For clarification, here's what it looks like now.

![image](https://github.com/owncast/owncast/assets/414923/f1ed2555-e233-4e66-bd9f-c9cf6c01d551)

**Comment by thezeeshann** [comment]
@gabek I think this one is better

**Comment by gabek** [comment]
> @gabek I think this one is better

Did you mean to attach something?

**Comment by thezeeshann** [comment]
> It does look better! But it would truncate for pretty much everyone, since accounts are almost always going to be longer than that. So if it's going to truncate for everybody, I'm not sure if that's the right design. What do you think?
> 
> For clarification, here's what it looks like now.
> 
> ![image](https://user-images.githubusercontent.com/414923/259463769-f1ed2555-e233-4e66-bd9f-c9cf6c01d551.png)

I'm just saying this design is better

**Comment by jerlinjosephh** [comment]
I would like to work on this issue. assign me this.

**Comment by imryanide** [comment]
Hii, is this still up for grabs?

**Comment by gabek** [comment]
@imryanide Absolutely! Thanks for wanting to be involved! If you have some ideas or mockups I'm happy to discuss ideas with you before making any changes if you'd like.

Here's a link to the corresponding components in Storybook: https://owncast.online/components/?path=/docs/owncast-components-followers-followers-collection--no-followers

**Comment by AMalfez** [comment]
That looks better.
Here are some of my ideas:
1. give some padding in icon and center everything inside each card.
2. can put a word limit on username so that design can look uniform.
3. can make card alignment responsive to adjust to different screen size.

**Comment by gabek** [comment]
The problem with a word limit on a username is it's always going to be one word.

Some will be `me@test.com` and others will be `coolperson42@social.exampledomain.com`. But no spaces and no logical way to wrap something like that. If we truncate it then it would be `coolperson42@soc...` and that's pretty useless to display at all in that case.

A responsive version what auto-adjusts the font size and image size to fit for different viewports is probably the answer.

Maybe on narrow screens it changes to a completely different layout. Something like

```
     [Avatar]
   Display Name
account@name.something.com
```

Just an idea so the image and the text isn't fighting for the narrow space.

**Comment by imryanide** [comment]
Yeah, I was thinking of the same, and maybe mess around with number of columns. I'll create a mockup first and share it over here once I'm done.

**Comment by gabek** [comment]
Hey there @imryanide! Just wanted to check in to see if you had any questions.

**Comment by imryanide** [comment]
Hey! I've given it some thought, and here is my plan, and change around the design a bit.

There are three main things I want to do :

1) Drop the email below the pfp so it has more space by default
2) Make the text font size responsive
3) Set a minimum card width 

Just as an example of what I imagine it to look like 

![image](https://github.com/owncast/owncast/assets/61741354/975f81a0-dde2-404d-af46-22a2d7dde0ab)

(sorry for the delay, was caught up with irl stuff)

Any thoughts?

**Comment by gabek** [comment]
I think the vertical stacked version does allow for more room and certainly helps the issue for narrow screens. What do you think about doing horizontal for most people, but it switching to your new vertical version when the viewport width hits a certain smaller breakpoint?

**Comment by imryanide** [comment]
Yep, that's exactly what I had in mind! will work on it

**Comment by freenandes** [comment]
we can try to shrink it even further to economize space. Even pushing the avatar up to gain some vertical.

![CleanShot 2024-01-26 at 17 41 30](https://github.com/owncast/owncast/assets/42041153/6ea4ecf5-f2e2-4f7e-9bbb-4bee31a43e62)

I wouldn't go as narrow as 160px wide. The middle row looks too broken. But going to a minimum of 240px (bottom row) for this component, we can easily fit this in mobile 320px wide devices.

With a flex layout, we can force it to go vertical when there is no space so that we don't occupy as much vertically, if we stretch the component too wide. But a maximum width for the component would be advisable so it doesn't look too weird

**Comment by freenandes** [comment]
I also removed the border and made it solid. The line work distracts too much and makes it much more rigid

**Comment by germainelee** [comment]
This sounds more like a responsive grid issue, than a change user card design layout to fit in smaller viewport width. Decreasing font size to fit text within user card and changing layout to vertical for smaller viewports would work but users might find a disconnected experience. I suggest that we stick to horizontal layout and maybe decrease follower username and profile pic size slightly.

Current implementation: The number of columns changes from 4 to 2 to 1 as the viewport width decreases. User card width dynamically changes and has a minimum width of 104px.

![image](https://github.com/owncast/owncast/assets/118464031/37db1226-7815-426b-878b-f9ba45c0d393)
^ image is screenshot at 901w, stream with chat

Proposed solution: The number of columns should decrease by 1, from 4 to 3 to 2 to 1, as the viewport width decreases. User card width dynamically changes and has a minimum width of 272px (or more depending on the "magic number"). So once the card width is decreased to that number, the number of columns would decrease.

![image](https://github.com/owncast/owncast/assets/118464031/9f46ec83-f338-4131-bd5e-ca07c818a763)
^ image is screenshot at 1440w, stream with chat

What do you think?

**Comment by gabek** [comment]
Oh I don't think moving away from a horizontal layout was in the cards for most viewports. But for mobile it's probably still necessary as it is now.

One thing that looks broken as it stands, is when one of those views is taller than the others. For example, the "fourthwall" one requires more vertical space, therefore it throws the layout off and looks wrong compared to the others that all have the same size.

Here's how it looks on mobile now, and is generally ok. 

![image](https://github.com/owncast/owncast/assets/414923/14a9d358-4d9e-48de-a771-17069fc88a94)

**Comment by germainelee** [comment]
FOURTHWALL user is not giving any issues in mobile. I believe this could be fixed if we increase the min width so that his card doesn't stick out in smaller viewport width. 

![Screenshot_20240609_171805_Chrome.jpg](https://github.com/owncast/owncast/assets/118464031/f256c3ae-2d24-4658-b98b-89cee69a44ec)

I can do a quick prototype in penpot to visualise the solution I have in mind if you don't mind.

**Comment by gabek** [comment]
Sorry for the confusion. I meant mobile is working fine for the vertical layout, so it shouldn't change to a horizontal one. It's only desktop that has multiple issues.

But if you'd like to prototype something, that'd be awesome!

**Comment by germainelee** [comment]
## Objective

- All cards have a fixed, uniform size
- Maintain responsiveness

## Acceptance Criteria

- [ ] when the name is 2-lines long in the card, the entire row should match that height while the other cards should be the same height. I don't want to see that big gap
- [ ] Design remains responsive across different devices and screen sizes. Number of columns should change gradually. For example, it could show four followers in 3 columns for 1440px wide screens and show four followers in 4 columns for 1920px screens. (Refer to video below for example)
- [ ] User card content alignment: Align content center
- [ ] minimum width of the user card is `min-width: 300px` or whatever number is better. This ensures that the username doesn't go 3 lines

## Expected Outcome
All follower cards should have a uniform size, providing a consistent and professional appearance, while still being responsive. See image below as example:

![Desktop Chat Open](https://github.com/owncast/owncast/assets/118464031/f0a40348-1f70-4fe2-8eda-00c43397ff70)

See video below for responsive card behavior, and ignore the card in second row (0:02) as it shouldn't stretch all the way to fill up the space. Remember, all follower cards must be of same width at all times and look uniform.

https://github.com/owncast/owncast/assets/118464031/21206723-b67a-4f21-beb6-34f6e6768fb7
<hr>

Hope that clarifies! Developers who are interested in working on this, see mockup in penpot (a similar design tool to Figma) for design specs.

**Comment by germainelee** [comment]
@gabek i'd like to work on this as it seems like a scss related and I'm more comfortable modifying css than tsx

**Comment by gabek** [comment]
Sure, feel free! Yup, it should just be css and html.
